images/upload-gitlab-failure-logs/upload_gitlab_failure_logs.py

```diff
@@ -35,6 +35,22 @@ def save(self, **kwargs):
         return super().save(**kwargs)
 
 
+class UnnecessaryJobPayload(Document):
+    timestamp = Date()
+
+    class Index:
+        name = "gitlab-unnecessary-jobs-*"
+
+    def save(self, **kwargs):
+        # assign now if no timestamp given
+        if not self.timestamp:
+            self.timestamp = datetime.now()
+
+        # override the index to go to the proper timeslot
+        kwargs["index"] = self.timestamp.strftime("gitlab-unnecessary-jobs-%Y%m%d")
+        return super().save(**kwargs)
```

Comment:
@mvandenburgh I'm not thrilled about having to duplicate most of this class, but I found little or no documentation on how the `Document` class is supposed to be used.  It would be nice to just be be able to change the index patterns since everything else is the same, but I'm not sure what bits of the class are used under the hood and how.  Some questions you may or may not know the answers to:

1. Why do we have the `timestamp` class member?  Do we want all instances to have the same value?
2. Is it intentional that there is also an instance member of the same name, but seemingly a different type?  One is a `opensearch_dsl.Date`, one is a `datetime` object.
3. How is the inner `Index` class used by the parent class?

Reply:
1/2) No, we don't want all instances to have the same value. I'm not sure why I put that there, maybe it was an oversight from a copy paste from an example like this https://elasticsearch-dsl.readthedocs.io/en/latest/#persistence-example. 

3) I believe the API of `opensearch_dsl` is the same as `elasticsearch_dsl`. So, the docs for the latter should help here - https://elasticsearch-dsl.readthedocs.io/en/latest/api.html#index